src/channel.cr

```diff
@@ -69,8 +69,20 @@ class Channel(T)
     Done   = 2
   end
 
+  private class SelectContextSharedState
+    @state : Atomic(SelectState)
+
+    def initialize(value : SelectState)
+      @state = Atomic(SelectState).new(value)
+    end
+
+    def compare_and_set(cmp : SelectState, new : SelectState) : {SelectState, Bool}
+      @state.compare_and_set(SelectState::Active, SelectState::Done)
+    end
```

Comment:
hmm, seems like these arguments aren't used anywhere... 🤔

Reply:
You're right. Fortunately, it is only used on those values 🙈.